### PR TITLE
Add OpenScPCA-nf docs page

### LINKED ISSUE #519: openscpca-nf docs page
This issue tracks adding a docs to the new section `ensuring-repro/openscpca-nf` (see #494) that introduces `OpenScPCA-nf`. To begin, we'll plan to write this as a single docs page with the following sections. As the doc gets built up, it make may more sense to split into multiple pages. 

Here are the planned sections:

- What is `openscpca-nf` and why do we have it?
- How/when do analyses move from `openscpca-analysis` to `openscpca-nf`?
- How does `openscpca-nf` use docker images?
- When is the `openscpca-nf` workflow run? It's run when a new `openscpca-nf` tag is made, and this is _in practice_ occurs with sufficient nf development and/or a new data release
- Explain how we run on both real and test data to produce results, which live in S3 (link to docs on obtaining results)

### PR DESCRIPTION
**Pull request by sjspielman**
Closes #519 

This PR adds a page for `OpenScPCA-nf`, and somewhat to my surprise (though perhaps in retrospect, this is not so surprising!) it's pretty short! In the original issue, I had envisioned each bullet point as a small section, but in practice most of those bullet points ended up being sentences, with a slightly longer explanation about results. I think the level of detail I included is about right for your average contributor, who mostly just needs to know that `OpenScPCA-nf` exists. I figure any more detailed docs live directly in the `nf` repo. But of course, let me know if/where you think these docs would benefit from some more detail. 

I made a few other changes as well:
- Ensured the headers in the reproducibility section match the names shown in the sidebar (any feedback on the wording I chose is helpful here too)
- Filled in relevant STUB links (there's a chance this actually wrapped up #504, but I'm not quite prepared to close that issue yet)

### CODE REVIEW

File: docs/ensuring-repro/openscpca-nf/index.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# The OpenScPCA Nextflow workflow
+
+In addition to the `OpenScPCA-analysis` repository, the Data Lab maintains [a separate repository called `OpenScPCA-nf`](https://github.com/AlexsLemonade/OpenScPCA-nf).
+The `OpenScPCA-nf` repository holds a [Nextflow workflow](https://www.nextflow.io/) which we use to generate results from mature `OpenScPCA-analysis` modules.
```
Comment (jashapiro) [inline docs/ensuring-repro/openscpca-nf/index.md:4]: This felt choppy, so I combined it into one sentence.
```suggestion
In addition to the `OpenScPCA-analysis` repository, the Data Lab maintains a separate [`OpenScPCA-nf` repository](https://github.com/AlexsLemonade/OpenScPCA-nf) that holds a [Nextflow workflow](https://www.nextflow.io/) which we use to generate results from mature `OpenScPCA-analysis` modules.
```

File: docs/ensuring-repro/openscpca-nf/index.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Specifically, once an analysis module has been completed, the Data Lab will port the analysis code into the `OpenScPCA-nf` repository and add it to the Nextflow workflow.
```
Comment (jashapiro) [inline docs/ensuring-repro/openscpca-nf/index.md:6]: I felt like this wasn't really making the previous sentence more specific, but rather adding context, so I removed the first word...
```suggestion
As analysis modules are completed, the Data Lab will port the analysis code into the `OpenScPCA-nf` repository and add it to the Nextflow workflow.
```

File: docs/ensuring-repro/openscpca-nf/index.md (hunk 1 above)
Comment (jashapiro) [reply docs/ensuring-repro/openscpca-nf/index.md:4]: In another place, we used the phrase "official results". I don't love that phrase, but I like the sentiment of it. We should say something like that here?

File: docs/ensuring-repro/workflows/run-module-gha.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
 To maintain module functionality over time, we use [GitHub Actions](https://docs.github.com/en/actions) (GHAs) to periodically run each module ("module testing GHA") with the goal of testing that the module code runs to completion without errors.
 
 !!! info
-    For more information about how we run modules to generate official results and OpenScPCA releases, please see our documentation on the `OpenScPCA-nf` workflow. <!-- openscpca-nf STUB_LINK -->
+    For more information about how we run modules to generate official results and OpenScPCA releases, please see our [documentation on the `OpenScPCA-nf` workflow](../openscpca-nf/index.md).
```
Comment (jashapiro) [inline docs/ensuring-repro/workflows/run-module-gha.md:6]: I don't love "official results" but I like the sentiment... Can we come up with another phrase that we can use more widely?

File: docs/getting-started/accessing-resources/getting-access-to-data.md [hunk 4]
```diff
@@ -145,7 +145,7 @@ This is generally the path you should use in your code.
 ## Accessing ScPCA module results
 
 As an analysis modules matures, the Data Lab will port the analysis to a separate repository called [`OpenScPCA-nf`](https://github.com/AlexsLemonade/OpenScPCA-nf).
-This repository holds a [Nextflow workflow](https://www.nextflow.io/) to reproducibly run and generate results from completed analysis modules.
+This repository [holds a Nextflow workflow](../../ensuring-repro/openscpca-nf/index.md) to reproducibly run and generate results from completed analysis modules.
```
Comment (jashapiro) [inline docs/getting-started/accessing-resources/getting-access-to-data.md:148]: I'm not sure we need the link directly to the repo here? maybe just to the docs page about it? And then maybe leave the Nextflow link? Or remove it completely.

**Review by jashapiro — COMMENTED** [review]
This looks good overall, and I agree that the level of detail seems correct.

I had a few wording and phrasing suggestions, and one place where I don't have a suggestion, but want to keep a sentiment... what should we call "official results"?

File: docs/ensuring-repro/workflows/run-module-gha.md (hunk 3 above)
Comment (sjspielman) [reply docs/ensuring-repro/workflows/run-module-gha.md:6]: Maybe "results for (intended for?) release", but that might a bit wordy?

Comment (sjspielman) [reply docs/ensuring-repro/workflows/run-module-gha.md]: I didn't quite land at a better phrase, but did this rephrase instead. What do you think? https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/582/commits/1fc1b0cb4e84282f99dcb4bb0000f42259878ba1

Worth noting the phrase "official results" doesn't appear anywhere else, but also this change I made doesn't really give a phrase to use more generally either

File: docs/getting-started/accessing-resources/getting-access-to-data.md (hunk 4 above)
Comment (sjspielman) [reply docs/getting-started/accessing-resources/getting-access-to-data.md]: I just removed the repo link on line 147

File: docs/ensuring-repro/workflows/run-module-gha.md [hunk 5]
```diff
@@ -3,7 +3,7 @@
 To maintain module functionality over time, we use [GitHub Actions](https://docs.github.com/en/actions) (GHAs) to periodically run each module ("module testing GHA") with the goal of testing that the module code runs to completion without errors.
 
 !!! info
-    For more information about how we run modules to generate official results and OpenScPCA releases, please see our documentation on the `OpenScPCA-nf` workflow. <!-- openscpca-nf STUB_LINK -->
+    For more information about how we run modules to generate OpenScPCA releases and associated results, please see our [documentation on the `OpenScPCA-nf` workflow](../openscpca-nf/index.md).
```
Comment (jashapiro) [inline docs/ensuring-repro/workflows/run-module-gha.md:6]: Since we don't run modules to generate releases, this still felt a bit off to me. How about this?
```suggestion
    For more information about how we run modules to generate workflow results associated with OpenScPCA data releases, please see our [documentation on the `OpenScPCA-nf` workflow](../openscpca-nf/index.md).
```

**Review by jashapiro — APPROVED** [review]
LGTM, with one little thought of another wording.

File: docs/ensuring-repro/workflows/run-module-gha.md (hunk 5 above)
Comment (sjspielman) [reply docs/ensuring-repro/workflows/run-module-gha.md:6]: I like it!

**Comment by sjspielman** [comment]
@jashapiro can I get a thumbs up (or not!) for this parallel change? https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/582/commits/447b59c85b52f59a18b53fff1c2bd81d1b525f67
